File: framework/languages/java/e2e/ResilienceLifecycle/Server/Provider/src/main/java/systems/zlink/e2e/resiliencelifecycle/provider/handlers/RuntimeErrorEvidenceHandler.java
package systems.zlink.e2e.resiliencelifecycle.provider.handlers;

import systems.zlink.e2e.resiliencelifecycle.provider.infrastructure.ScenarioState;
import systems.zlink.framework.monitoring.ZLinkRuntimeErrorEvent;
import systems.zlink.framework.monitoring.ZLinkRuntimeEventHandler;

public final class RuntimeErrorEvidenceHandler
    implements ZLinkRuntimeEventHandler<ZLinkRuntimeErrorEvent> {
    private final ScenarioState state;

    public RuntimeErrorEvidenceHandler(ScenarioState state) {
        this.state = state;
    }

    @Override
    public void handle(ZLinkRuntimeErrorEvent event) {
        state.record("RuntimeError", event.event().name() + "/" + event.callbackName());
    }
}

File: framework/languages/java/e2e-kotlin/ResilienceLifecycle/Server/Provider/src/main/kotlin/systems/zlink/e2e/kotlin/resiliencelifecycle/Handlers/RuntimeErrorEvidenceHandler.kt
package systems.zlink.e2e.kotlin.resiliencelifecycle

import systems.zlink.framework.monitoring.ZLinkRuntimeErrorEvent
import systems.zlink.framework.monitoring.ZLinkRuntimeEventHandler

class RuntimeErrorEvidenceHandler(
    private val state: ScenarioState,
) : ZLinkRuntimeEventHandler<ZLinkRuntimeErrorEvent> {
    override fun handle(event: ZLinkRuntimeErrorEvent) {
        state.record(
            "RuntimeError",
            "${event.event()}/${event.callbackName()}/${event.exceptionType()}/${event.message().orElse("")}",
        )
    }
}
